compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -542,12 +545,21 @@ private Path emitGraalExecutable(Path executableFilePath) {
         String nativeImageName;
         String[] command;
         Project project = this.packageContext().project();
+        String nativeImageCommand = System.getenv("GRAALVM_HOME");
+
+        if (nativeImageCommand == null) {
+            throw new ProjectException("unable to create native image. The environment variable GRAALVM_HOME " +
+                    "contains an empty string.");
+        }
+
+        nativeImageCommand += File.separator + BIN_DIR_NAME + File.separator
```

Comment:
If possible it is better to check whether `native-image.exe` or `native-image` is located in the bin directory before executing the command. Else this will  ends up in the same error message which is not descriptive :
```
error: unable to create native directory:Cannot run program "native-image": error=2, No such file or directory
```

If the native-image tool is not found, we could return an error like this :
```
Cannot find <executableName> in the GRAALVM_HOME. Install it using: gu install native-image
```

Reply:
Added this check after the environment variable is resolved. Will print an error if native isnt installed or if it cant find the executable